### PR TITLE
Add autoTable to jsPDF node dist files

### PR DESCRIPTION
**Pull request by simonbengtsson**
What do you think of this? @mmghv @rmdias

### CODE REVIEW

**Comment by mmghv** [comment]
I don't like the idea of trying to apply the plugin to all `jsPDF` modules in all cases, adding the extra overhead and size of loading all 3 modules in all times (especially in the browser) which 2 of them won't be used.

What's wrong with giving the user the power to choose which `jsPDF-AutoTable` version he wants? .. something like that :

```js
// browser version
const jsPDF = require('jspdf')
require('jspdf-autotable')

// node version
const jsPDF = require('jspdf/dist/jspdf.node.debug.js')
require('jspdf-autotable/dist/jspdf.plugin.autotable.node.js')
```

Then you could make `jspdf.plugin.autotable.node.js` import and inject `autoTable` on  both `jspdf/dist/jspdf.node.min.js` and `jspdf/dist/jspdf.node.debug.js`.

Or you could let the user inject `autoTable` himself in whatever module he wants :

```js
const jsPDF = require('jspdf/dist/jspdf.node.debug.js')
const autoTable = require('jspdf-autotable/dist/jspdf.plugin.autotable.node.js')
autoTable(jsPDF)
```

In the latter choice you will not import any `jsPDF` modules yourself, rather export the `applyApi()` function only.

**Comment by simonbengtsson** [comment]
What do you mean with `extra overhead and size`?

I like the idea of exporting the applyApi function, added a commit for that now 👌

File: src/main.ts [hunk 1]
```diff
@@ -3,7 +3,39 @@
...
+  }
+} catch (err) {}
+
+try {
+  let jsPDF = require('jspdf/dist/jspdf.node.debug')
+  if (jsPDF) {
+    applyApi(jsPDF)
+    appliedApi = true
+  }
+} catch (err) {}
```
Comment (mmghv) [inline src/main.ts]: Here you are trying to load all three `jspdf` dist modules, If the developer is building for the browser and is using a build setup (like webpack) then his bundle will include all three `jspdf` modules (when two of them are for node) adding extra overhead and increasing the bundle size for no reason.

I actually tested it myslef, bundle size increased from `327KB` to `850KB` (minified).

So we should only `require('jspdf')` for the browser version of the plugin, and for the node version we don't `require()` any module and let the developer `applyPlugin()` himself on the `jspdf` module that he imported.

**Comment by simonbengtsson** [comment]
True. If you are building for the browser you would have to mark the node dist files as external for them to not be bundled which certainly would be unexpected. Will update pr 👌

File: src/main.ts (hunk 1 above)
Comment (simonbengtsson) [reply src/main.ts]: 💪

**Comment by simonbengtsson** [comment]
Since there are a minified and a debug version of the nodejs dist files for jspdf I think we should not provide any additional dist files for autotable. Hopefully the jspdf nodejs dist files will be removed in the future and support for nodejs provided out of the box in jspdf. For now I'll merge and release a new version simply exporting the applyPlugin method.

Let me know if you have any additional thoughts @mmghv

**Comment by mmghv** [comment]
Looks great.
good idea for exporting only one version of the plugin, trying to import the default version of `jspdf` in nodejs isn't bad and the extra size increased for importing it is not a problem in the server side.

I would suggest renaming `applyApi()` to `applyPlugin()` for consistency, then in `main.ts` you would simply :

```js
export {applyPlugin}
```

**Comment by simonbengtsson** [comment]
Nice! Updated it now. Didn't know of the  `export {applyPlugin}` syntax.

**Pull request merged**